### PR TITLE
Revert change to reuse artifacts for Windows

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**: I'm not entirely sure what happened, but while I was testing #3455, I noticed the images were getting rebuilt and that the docker manifest os.Version didn't contain the full version. However, I'm now seeing the opposite. It's possible I was testing with an old version of cloud-provider-azure and the image build/push has changed. In any case, what we had before seems to be correct, so this reverts the change made in #3455 to the reuse_artifacts function so we're not repushing images unless it's necessary.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: scripts/ci-build-azure-ccm.sh [hunk 1]
```diff
@@ -82,7 +82,8 @@ can_reuse_artifacts() {
         fi
     done
 
-    if ! docker manifest inspect "${REGISTRY}/${CNM_IMAGE_NAME}:${IMAGE_TAG_CNM}" | grep -q "\"os.version\": \"${WINDOWS_IMAGE_VERSION}\""; then
+    FULL_VERSION=$(docker manifest inspect mcr.microsoft.com/windows/nanoserver:${WINDOWS_IMAGE_VERSION} | jq -r '.manifests[0].platform["os.version"]')
+    if ! docker manifest inspect "${REGISTRY}/${CNM_IMAGE_NAME}:${IMAGE_TAG}" | grep -q "\"os.version\": \"${FULL_VERSION}\""; then
```
Comment (CecileRobertMichon) [inline scripts/ci-build-azure-ccm.sh:86]: /hold

Comment (CecileRobertMichon) [reply scripts/ci-build-azure-ccm.sh]: /hold cancel

**Review by jackfrancis — APPROVED** [review]
/lgtm
/approve

**Comment by jackfrancis** [comment]
/hold

/test pull-cluster-api-provider-azure-windows-containerd-upstream-with-ci-artifacts

@CecileRobertMichon feel free to cancel hold if we get green signal on the above test

**Pull request merged**